pkg/operator/apiserver/controller/apiservice/apiservice_controller.go

```diff
@@ -152,11 +168,28 @@ func (c *APIServiceController) sync(ctx context.Context, syncCtx factory.SyncCon
 	return err
 }
 
-func (c *APIServiceController) syncAPIServices(ctx context.Context, apiServices []*apiregistrationv1.APIService, recorder events.Recorder) error {
+func (c *APIServiceController) syncDisabledAPIServices(ctx context.Context, apiServices []*apiregistrationv1.APIService) error {
 	errs := []error{}
-	var availableConditionMessages []string
 
 	for _, apiService := range apiServices {
+		if _, err := c.apiservicelister.Get(apiService.Name); err == nil {
```

Comment:
should we also consider a case in which a service has been already marked for deletion but hasn't been deleted?
(I think that would require checking the `DeletionTimestamp` field)

Reply:
That should be idempotent. Deleting an object that's in the process of deletion should be a no-op. While `Get` should still return the object without an error.

Reply:
It is idempotent but we will hammer the API unnecessary. Does it make sense?

Reply:
I agree this will reduce the toll. On the other hand I wonder if checking the `DeletionTimestamp` is something we already do in other controller or whether this is a novel improvement. If the latter, I prefer to refactor the code more and implement generic methods like DeleteOnlyIfPresent. Probably automatically generated for all our APIs. So we don't re-invent the check all over the place. What do you think?

Reply:
I don't think it is novel approach. For example, it is being used in:

- https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/auth/authenticator/token/bootstrap/bootstrap.go#L107

- https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/serviceaccount/legacy_serviceaccount_token_cleaner.go#L147

I can try to find more examples if you want.

Reply:
I was wondering if we have an example in our operators. Nevertheless, checking for the `DeletionTimestamp` is easy to implement. Will do.